File: app/src/main/java/com/rdev/tryp/intro/manager/AccountManager.java
package com.rdev.tryp.intro.manager;

import com.rdev.tryp.model.RealmUtils;
import com.rdev.tryp.utils.PreferenceManager;

/**
 * Created by Alexey Matrosov on 06.03.2019.
 */
public class AccountManager {
    private static AccountManager instance;
    private final static String USER_ID_KEY = "USER_ID_KEY";

    private int userId;

    public static AccountManager getInstance() {
        if (instance == null)
            instance = new AccountManager();
        instance.userId = PreferenceManager.getInt("USER_ID_KEY");
        return instance;
    }

    public int getUserId() {
        return userId;
    }

    public boolean isUserSignIn(){
        return userId != -1;
    }

    public void signIn(int userId){
        this.userId = userId;
        PreferenceManager.setInt(USER_ID_KEY, userId);
    }

    public void signOut(){
        userId = -1;
        PreferenceManager.setInt(USER_ID_KEY, userId);

        RealmUtils realm = new RealmUtils(null);
        realm.clear();
    }
}

File: app/src/main/java/com/rdev/tryp/intro/manager/AccountManager.kt
package com.rdev.tryp.intro.manager

import android.content.Context
import com.rdev.tryp.model.RealmUtils
import com.rdev.tryp.utils.PreferenceManager

/**
 * Created by Alexey Matrosov on 06.03.2019.
 */


class AccountManager {

    var userId: Int = -1

    var isUserSignIn = userId != -1

    fun signIn(userId: Int) {
        this.userId = userId
        PreferenceManager.setInt(USER_ID_KEY, userId)
    }

    fun signOut(context: Context) {
        userId = -1
        PreferenceManager.setInt(USER_ID_KEY, userId)
        val realm = RealmUtils(context, null)
        realm.clear()
    }

    companion object {
        private var instance: AccountManager? = null
        private const val USER_ID_KEY = "USER_ID_KEY"

        fun getInstance(): AccountManager? {
            if (instance == null)
                instance = AccountManager()
            PreferenceManager.getInt(USER_ID_KEY)?.let { userId ->
                instance?.userId = userId
            }
            instance?.isUserSignIn = instance?.userId != -1

            return instance
        }
    }

}